### PR TITLE
Add branch deployment

### PR DESCRIPTION
**Pull request by elenagarrone**
# Description

This PR add branch deployment to GitHub Pages:

- remove `actions` folder and move the scripts in workflow-scripts for consistency
- add scripts for `deployment`, `un-deployment` and `adding a comment to a PR`
- update `master` pipeline to use the new `deployment` script
- add new `workflow_dispatch` workflow to handle branch deployment - the deployment will be triggered manually so that one of the Core or Regular Contributors can review the code before deploying it. It also checks that the `checks` on the PR have passed successfully
- add new workflow to handle the un-deployment of github pages
- add `pre-push` hook to check format of branch name
- remove Netlify configuration
- update documentation

**This will need to be merged before being able to test it.**

### CODE REVIEW

File: .github/workflow-scripts/deploy-branch.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    owner,
+    repo,
+    check_run_id: pullRequest.head.sha
+  });
+
+  return listCheckRuns.every(checkRun => checkRun.conclusion === 'success');
+}
+
+async function deploy(branch, exec) {
+  let docsPath = branch === 'master' ? './docs' : `./docs/${branch}`;
```
Comment (pauleustice) [inline .github/workflow-scripts/deploy-branch.js:46]:
```suggestion
  const docsPath = branch === 'master'
    ? './docs'
    : `./docs/${branch}`;
```

Comment (elenagarrone) [reply .github/workflow-scripts/deploy-branch.js]: Updated 👍

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -90,6 +91,11 @@ To ensure that changes come from an entrusted source all commits must be [signed
 2. [Adding a new GPG key to your GitHub account](https://help.github.com/en/articles/adding-a-new-gpg-key-to-your-github-account)
 3. [Signing commits](https://help.github.com/en/articles/signing-commits)
 
+## Branch naming convention
+
+Branches should be in `kebab-case` format. 
+The repository is set up with a `pre-push` hook that will prevent the user to push if the name doesn't match the correct format.
```
Comment (pauleustice) [inline docs/CONTRIBUTING.md:97]:
```suggestion
The repository is set up with a `pre-push` hook that will prevent the user from pushing if the branch name doesn't match the correct format.
```

File: .github/workflow-scripts/deploy-branch.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  const docsPath = branch === 'master' ? './docs' : `./docs/${branch}`;
+
+  console.info('ℹ️ Commencing storybook gh-pages deploy');
+
+  console.info('ℹ️ Running storybook build');
+  // gh-pages only works in the root directory, or '/docs'
+  let commandsArray = ['run', 'build-storybook', '--', '-o', docsPath];
+  if (branch !== 'master') {
+    // -c: change the folder that Storybook uses
+    commandsArray = [...commandsArray, '--', '-c', docsPath];
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply .github/workflow-scripts/deploy-branch.js:55]: That's a good point - I will test it and update it 👍

[the remark this answers is not in the document]
Comment (elenagarrone) [reply .github/workflow-scripts/deploy-branch.js]: Updated 👍

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (elenagarrone) [reply docs/CONTRIBUTING.md]: Updated 👍

File: .husky/pre-push [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+# Branch names must be kebab-case
+
+npx enforce-branch-name '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$'
```
Comment (owensgit) [inline .husky/pre-push:6]: Do we need to add `enforce-branch-name` to package.json, or does it not matter as we're using `npx`?

**Review by owensgit — APPROVED** [review]
Looks great! Just one comment from me, but approved anyway.

File: .husky/pre-push (hunk 4 above)
Comment (elenagarrone) [reply .husky/pre-push:6]: Yes, that's right 🙂  Because we are using `npx` there is no need to install the package 👍

**Review by pauleustice — APPROVED** [review]
Brilliant work 👍

**Review by owensgit — APPROVED** [review]

**Pull request merged**